### PR TITLE
Adjust deprecation API slightly to prepare for public preview

### PR DESCRIPTION
**Pull request by joelverhagen**
- No-op the last edit update if there are no changes (reduces V3 load)
- Require user agent and include the full UA as a custom dimension in App Insights
- Return 400 not 404 when the alternate package is missing 
- Require user agent for the deprecation endpoint
- These changes are backwards compatible with the known integrations with the Deprecation API

Depends on https://github.com/NuGet/NuGetGallery/pull/9505.

Align the API with the design changes described here: https://github.com/NuGet/Engineering/pull/4709 (internal PR now, will be public after public preview begins).

### CODE REVIEW

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -529,6 +529,15 @@
       <add initializationPage="/api/health-probe"/>
     </applicationInitialization>
   </system.webServer>
+  <!-- Don't use the default error pages for API endpoints. -->
```
Comment (zhhyu) [inline src/NuGetGallery/Web.config:532]: Merge recent DEV changes?

**Review by zhhyu — APPROVED** [review]

File: src/NuGetGallery/App_Data/Files/Content/flags.json [hunk 2]
```diff
@@ -68,8 +68,8 @@
       "Domains": []
     },
     "NuGetGallery.ManageDeprecationApi": {
-      "All": false,
-      "SiteAdmins": true,
+      "All": true,
```
Comment (zhhyu) [inline src/NuGetGallery/App_Data/Files/Content/flags.json:71]: Do we plan to enable it immediately when the rollout is done?

Comment (joelverhagen) [reply src/NuGetGallery/App_Data/Files/Content/flags.json:71]: This is for local development only. I think it's best to default it on for local development.

File: src/NuGetGallery/Web.config (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery/Web.config]: Yup, thanks.

**Pull request merged**